src/colorizer/utils/data_utils.ts

```diff
@@ -222,3 +222,21 @@ export function getIntervals(values: number[]): [number, number][] {
   }
   return intervals;
 }
+
+export function hasPropertyChanged<T extends Record<string, unknown>>(
```

Comment:
i am sad that you actually had to implement this function

Reply:
Should I not have???

Reply:
mostly because I think of React is supposed to do this, and also I'm surprised that there isn't a great library version of this..

Reply:
Mm, React does do this but we're not in React-land so I can't use hooks for this... There probably is but I haven't looked